modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/checkpoint/CheckpointEntry.java

```diff
@@ -356,9 +356,12 @@ private void initIfNeeded(
 
                         grpStates = remap(stateRec);
                     }
-                    else
+                    else {
                         initEx = new IgniteCheckedException(
                             "Failed to find checkpoint record at the given WAL pointer: " + ptr);
+
+                        throw initEx;
```

Comment:
As I understand you can skip the assignment of initEx here, because if you just throw an exception from here, initEx will be assignmented in the catch block.

Reply:
Done.